core/chains/evm/headtracker/heads.go

```diff
@@ -60,6 +62,37 @@ func (h *heads) Count() int {
 	return len(h.heads)
 }
 
+func (h *heads) MarkFinalized(finalized common.Hash) bool {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	if len(h.heads) == 0 {
+		return false
+	}
+
+	// copy slice as we are going to modify it
+	newHeads := make([]*evmtypes.Head, len(h.heads))
```

Comment:
Shouldn't we copy this to h.heads ? It seems like we do all the work but we don't actually update the original structure.